core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/rules/SnowflakeCallMapper.scala

```diff
@@ -648,4 +658,24 @@ class SnowflakeCallMapper extends ir.CallMapper with ir.IRHelpers {
     irSortArray
   }
 
+  private def parse_last_day(args: Seq[ir.Expression]): ir.Expression = {
+    if (args.length == 1) {
+      return ir.LastDay(args.head)
+    }
+
+    val datePart = args(1) match {
+      case ir.StringLiteral(part) => part.toUpperCase()
+      case ir.Column(_, part) => part.toString.toUpperCase()
```

Comment:
This may throw an (uncaught) `MatchError` if `arg(1)` is neither `StringLiteral` nor `Column`, please either handle the case or raise a `RemorphError`